numpy/core/tests/test_datetime.py

```diff
@@ -471,6 +471,19 @@ def test_timedelta_nat_format(self):
         # gh-17552
         assert_equal('NaT', '{0}'.format(np.timedelta64('nat')))
 
+    def test_format(self):
+        vs = "2021-12-12T10:10:10"
+        v = np.datetime64(vs)
+        with assert_raises_regex(
+            ValueError,
+            "Unknown format code 'f' for object of type 'str'"
```

Comment:
Avoid being too specific when matching error messsages
```suggestion
            "Unknown format code"
```